lfs/scanner.go

```diff
@@ -517,3 +528,149 @@ func lsTreeBlobs(ref string) (chan TreeBlob, error) {
 
 	return blobs, nil
 }
+
+// ScanUnpushed scans history for all LFS pointers which have been added but not pushed to any remote
+func ScanUnpushed() ([]*WrappedPointer, error) {
+
+	start := time.Now()
+	defer func() {
+		tracerx.PerformanceSince("scan", start)
+	}()
+
+	pointerchan, err := logUnpushedSHAs()
+	if err != nil {
+		return nil, err
+	}
+	pointers := make([]*WrappedPointer, 0, 10)
+	for p := range pointerchan {
+		pointers = append(pointers, p)
+	}
+	return pointers, nil
+}
+
+// logUnpushedSHAs scans history for all LFS pointers which have been added but not pushed to any remote,
+// return progressively in a channel
+func logUnpushedSHAs() (chan *WrappedPointer, error) {
+	logArgs := []string{"log",
+		"--branches", "--tags", // include all locally referenced commits
+		"--not", "--remotes", // but exclude everything reachable from any remote
+	}
+	// Add standard search args to find lfs references
+	logArgs = append(logArgs, logLfsSearchArgs...)
```

Comment:
Does it need the global `logLfsSearchArgs`? I don't see it referenced anywhere else.